adjust platform compatibility for qnn backend

h-friederich opened the pull request:
Summary: The Qualcomm dependencies are only available for Android, so let's mark the target accordingly.

Differential Revision: D75139951

h-friederich commented:
@pytorchbot label "release notes: none"

cccclai reviewed (approved):
Internal only change

h-friederich committed:
adjust platform compatibility for qnn backend (#11179)

Summary:
The Qualcomm dependencies are only available for Android, so let's mark the target accordingly.

Pull Request resolved: https://github.com/pytorch/executorch/pull/11179

Test Plan:
Imported from GitHub, without a `Test Plan:` line.

Rollback Plan:

Differential Revision: D75139951

Pulled By: h-friederich
